Assets/Scripts/CreatureStats.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class CreatureStats : MonoBehaviour
+{
+    public Dictionary<StatType, float> _stats;
+
+    void Start()
+    {
+        // Choose stats for creature
+        // _stats = new Dictionary<StatType, float>()
+        // {
+        //     { StatType.Health, 100f },
+        //     { StatType.MaxHealth, 100f },
+        //     { StatType.ArmorPercentage, 0f },
+        // };
+    }
+
+    public float ValidateStat(StatType statType, float statValue)
+    {
+        if (statType == StatType.Health)
+        {
+            return Mathf.Clamp(statValue, 0, _stats[StatType.MaxHealth]);
+        }
+        else if (statType == StatType.ArmorPercentage || statType == StatType.ReflectionPercentage)
+        {
+            return Mathf.Clamp(statValue, 0, 1);
+        }
+        return statValue;
+    }
+
+    public void ChangeStats(StatType statType, float statValue)
+    {
+        if (_stats.ContainsKey(statType))
+        {
+            _stats[statType] = ValidateStat(statType, statValue);
+            Debug.Log($"{statType} set to {_stats[statType]}");
+        }
+        else
+        {
+            Debug.LogWarning($"Stat {statType} not found!");
+        }
+    }
+
+    public void AddToStat(StatType statType, float valueToAdd)
```

Comment:
Most likely we will be satisfied with just one method. And it, again, can be inside other class.

Reply:
We can have items that reset the value of stat to 0. I think this is enough for our program to work how intended and intuitively develop further.